### DISCUSSION TITLE
Milvus pods on EKS do not initialize (initialize forever)

### DISCUSSION THREAD

**Discussion by ramialt** [milvus-io/milvus#25221] [Q&A and General discussion] [answered]
![Screenshot 2023-06-28 at 6 12 26 PM](https://github.com/milvus-io/milvus/assets/56455158/b49dc715-3ff2-4e98-916e-ff602a06a368)

Pulsar pods in particular never get out of pod initializing, and etcd is stuck in pending which causes errors in other pods.
After checking logs of these pods using `kubectl logs milvus-pulsar-pulsar-init-jz2hp -c wait-zookeeper-ready`,
I find that the zookeeper cannot be found:
```
** server can't find milvus-pulsar-zookeeper-2.milvus-pulsar-zookeeper.default: NXDOMAIN

Server:         10.100.0.10
Address:        10.100.0.10#53
<repeated>
```
From my understanding the 10.100.0.10 is likely one of the coredns pods? I have tried sh-ing into the pods to test nslookup, but that command was not installed, and I could not install it after an attempt. Maybe someone can provide the proper security group setup inbound/outbound rules? Though I'm not sure why those would be different given my deployment.

#### Deployment details:
Am deploying to EKS with S3 using Helm.
##### **`values.yaml` changes:**
```
helm upgrade --install 
--set cluster.enabled=true --set minio.enabled=False 
--set externalS3.enabled=true --set externalS3.useIAM=True 
--set externalS3.host='s3.REGION.amazonaws.com' --set externalS3.port=443 
--set externalS3.useSSL=true --set externalS3.bucketName='milvus-bucket' 
--set service.type=LoadBalancer milvus milvus/milvus
```

Since I am using IAM, I attempted to follow this guide [Using IAM roles to allow the Pods in AWS EKS to read the AWS S3 bucket](https://levelup.gitconnected.com/using-iam-roles-to-allow-the-pods-in-aws-eks-to-read-the-aws-s3-bucket-be493fbdda84). On the "Create a Service Account" step I noticed that there is a `serviceAccount` field in `values.yaml`, which I modified like so:
```
serviceAccount:
  create: true
  name: milvus-service-account
  annotations:
    eks.amazonaws.com/role-arn: <the account role arn>
  labels:
```

I am not sure if this is the correct service account setup, but when doing `kubectl describe milvus-datanode` I do see the aws-iam-token volume and the new env vars AWS_WEB_IDENTITY_TOKEN_FILE, AWS_ROLE_ARN, etc.

#### Logs
[datacoord.log](https://github.com/milvus-io/milvus/files/11901165/datacoord.log)
[indexcoord.log](https://github.com/milvus-io/milvus/files/11901166/indexcoord.log)
[proxy.log](https://github.com/milvus-io/milvus/files/11901167/proxy.log)
[querycoord.log](https://github.com/milvus-io/milvus/files/11901169/querycoord.log)
[rootcoord.log](https://github.com/milvus-io/milvus/files/11901170/rootcoord.log)

Would GREATLY appreciate some help! I've been struggling on this for a while.

**Comment by LoveEachDay** [comment]
@ramialt We rely on the default storageclass to provision a volume for milvus. Could you check whether there is a default storageclass like this:
```
kubectl get storageclass
```

It should output something with `(default)` .

If not, you'd setup a default storageclass for your kubernetes cluster.

**Reply by ramialt** [reply]
I get `(default)` output:
```
NAME            PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
gp2 (default)   kubernetes.io/aws-ebs   Delete          WaitForFirstConsumer   false                  8h
```
This does not change after 5 minutes (usually the pod status looks like above screenshot status in <30s).
Interesting that it says age is 8 hours, thought I would have deleted everything after doing helm uninstall. I guess this is created from `milvus_cluster.yaml` step?

**Reply by ramialt** [reply, marked as answer]
Error seems to have been fixed after recreating cluster from scratch. Hard to remember for sure, but I believe I messed up when doing `eksctl create cluster -f milvus_cluster.yaml` step somehow and the ebs addon did not work or something. Thank your for your help

**Comment by ramialt** [comment]
The error suggests it is looking for an milvus-pulsar-zookeeper-2 node when there is only a milvus-pulsar-zookeeper-0 node. Could this assumption be the root of the issue?
`server can't find `**`milvus-pulsar-zookeeper-2`**`.milvus-pulsar-zookeeper.default`
